packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -213,17 +213,32 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> implements PredictiveB
   /// It defaults to `true`.
   bool willDisposeAnimationController = true;
 
+  /// Returns true if [dispose] has been called.
+  ///
+  /// After [dispose] is called, this object can no longer be used.
+  ///
+  /// This method only works if assert is enabled. Otherwise it always returns
+  /// false.
+  @protected
+  bool debugIsDisposed() {
+    bool disposed = false;
+    assert(() {
+      disposed = _transitionCompleter.isCompleted;
```

Comment:
this seems to be a weird way to determine dispose. Should this be in the base route class and based on the value _disposeCompleter.isCompleted ?

Reply:
This is simply a refactor, see `createAnimationController` before the PR (the left column) and other methods. I refactored this because `_transitionCompleter` isn't public and I wanted this extra check in my overridden `createSimulation`.

Reply:
I am fine with having this extra public method, but my concern is that disposed should be mapped to _disposeCompleter not _transitionCompleter. If you really meant to use transitionCompleter, I think this method needs a better name

Reply:
The reason why I chose this name mostly comes from its assert message:
```dart
assert(!_transitionCompleter.isCompleted, 'Cannot reuse a $runtimeType after disposing it.');
```
I'm happy to accept other names, but then I think this assertion message would be wrong.

Considering the completer's Future has the following comment:
```dart
  /// This future completes only once the transition itself has finished, after
  /// the overlay entries have been removed from the navigator's overlay.
  ///
  /// This future completes once the animation has been dismissed. That will be
  /// after [popped], because [popped] typically completes before the animation
  /// even starts, as soon as the route is popped.
```

What do you think of  the name `debugTransitionFinished`? (We might still want to change the assert message though...)

Reply:
the debugTransitionFinished sounds good to me